PYIC-2439: Create a CredentialIssuerConfigService that inherits methods from ConfigService.

Danomanic opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed
- Create a CredentialIssuerConfigService that inherits methods from ConfigService.

<!-- Describe the changes in detail - the "what"-->

### Why did it change

<!-- Describe the reason these changes were made - the "why" -->

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-2439](https://govukverify.atlassian.net/browse/PYIC-2439)


[PYIC-2439]: https://govukverify.atlassian.net/browse/PYIC-2439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
PYIC-2439: Create a CredentialIssuerConfigService that inherits methods from ConfigService.

Commit:
PYIC-2439: Remove unnecessary qualification.

Commit:
PYIC-2439: Remove unused method from ConfigService.

Commit:
PYIC-2439: Remove calls from core library to getCredentialIssuer in ConfigService.

Co-authored-by: Daniel Pomfret <daniel.pomfret@digital.cabinet-office.gov.uk>
Co-authored-by: Kerr Rainey <kerr.rainey@digital.cabinet-office.gov.uk>

Commit:
PYIC-2439: Fix KBV Issuer

blakeyp commented on lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java:
```diff
@@ -113,6 +114,18 @@ public String getSsmParameter(ConfigurationVariable configurationVariable, Strin
+        if (recursive) {
+            return ssmProvider.recursive().getMultiple(path);
+        } else {
+            return ssmProvider.getMultiple(path);
```
Maybe just have two separate methods `getSsmParameters` and `getSsmParametersRecursive` to avoid the boolean flag arg?

kerrr committed:
BAU: rename ConfigurationService to ConfigService

We frequently encounter issues where line length inhibits readability, especially when it forces wrapping. This change to a less verbose version does not result in any ambiguity and helps with line length issues.

This is a simple rename refactor, with no behavioral changes.
